### ISSUE TITLE
Stop/Start no longer refreshing secrets from KeyVault references

### ISSUE THREAD

**Issue by 23min** [Azure/Azure-Functions#1855]
My UC is that some secrets in the keyvault periodically are updated. Until now, I have been able to use the Azure Management api to stop and start a Function App, and the environment variables would be updated with new values.

Since recently, stop/start has no effect. I receive stale values from keyvault references.

If I make a small change in the App Settings by hand, like adding a new setting, fresh values are obtained from keyvault references.

Currently on 
```
  "version": "3.0.15371.0",
  "versionDetails": "3.0.15371 Commit hash: 67907e30b20e9f1bb21bb1f4bb896619cc380636",
  "platformVersion": "92.0.7.74",
```

Is this a regression?

If this is by design (performance for cold start? etc.), what is the correct way to programmatically trigger a reload of settings in a Function App?

**Comment by balag0** [comment]
Yes restarting the app should have worked as would updating an app setting / site config.
Could you share the app name and the time when the restart was done.
https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately

**Comment by 23min** [comment]
One of the Function Apps:
```
2021-03-02T08:20:57.289 [Information] Host Status: {
  "id": "xxxxxxxxxx",
  "state": "Running",
  "version": "3.0.15371.0",
  "versionDetails": "3.0.15371 Commit hash: 67907e30b20e9f1bb21bb1f4bb896619cc380636",
  "platformVersion": "92.0.7.74",
  "instanceId": "d40e498bf73a1d6d1c1f9a60aa1ef8799ab900a5bbf969760e71766610a6e935",
  "computerName": "RD28187896FFA0",
  "processUptime": 1680058
}
```

**Comment by balag0** [comment]
@23min Thanks. This information above was not unique enough to identify the app. Can you try one of the 2 options listed at 
https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately

**Comment by 23min** [comment]
Thanks,
The app is in North Europe.
Neither step 1 nor step 2 in the page you reference seem to work. 
I extracted the function Id:
```
2021-03-02T08:34:05.376 [Information] Executing 'ServiceBusMetadata' (Reason='This function was programmatically called via the host APIs.', Id=24f44227-26f6-497f-a4d9-555543b4cba3)
2021-03-02T08:34:05.377 [Information] Executed 'ServiceBusMetadata' (Succeeded, Id=24f44227-26f6-497f-a4d9-555543b4cba3, Duration=0ms)
2021-03-02T08:34:05.376 [Information] Executing 'ServiceBusMetadata' (Reason='This function was programmatically called via the host APIs.', Id=24f44227-26f6-497f-a4d9-555543b4cba3)
2021-03-02T08:34:05.377 [Information] Executed 'ServiceBusMetadata' (Succeeded, Id=24f44227-26f6-497f-a4d9-555543b4cba3, Duration=0ms)
```
Alternatively, I can send the name of the Function App to you via email.

**Comment by 23min** [comment]
Tested again, with a different Function App, also in North Europe.

(I added setting WEBSITE_USE_PLACEHOLDER=0 to see if it would make a difference)

1. Updated KeyVault secret
2. Automated stop/start via Azure Management API. Result: secret has old value
3. Manual Stop/Start via portal @2021-03-03T13:17:51.1955912Z. Result: secret has old value.
4. Manual Restart via portal @2021-03-03T13:45:53.6297Z. Result: secret has old value. 
5. Manually updating a setting in the Configuration blade for the Function App. Result: secret has **NEW** value!

InvocationId: 6f4ec1e2-1a40-4313-8e45-e539b009def8

![image](https://user-images.githubusercontent.com/1111124/109819636-92ebc280-7c34-11eb-9d51-c96c1dec6785.png)

**Comment by v-anvari** [comment]
cc @balag0

**Comment by 23min** [comment]
I've had contact with Azure support. 

"Unfortunately, at this point the Stop/Start behavior not working is “by design” based on changes done on the Platform level."

"Stop Start was never defined behavior for refreshing versionless key vault references. We've recently rolled out our defined behavior for versionless“.

“If a version is not specified in the reference, then the app will use the latest version that exists in Key Vault. When newer versions become available, such as with a rotation event, the app will automatically update and begin using the latest version within one day. Any configuration changes made to the app will cause an immediate update to the latest versions of all referenced secrets.”

The best approach is to initiate rotation via a configuration change/site update.

"A PUT action against the site resource, or the siteconfig resource should be good to trigger the action goal is that the gesture conveys the intent for the site's configuration to update, versus a restart behaves more indeterministically (as there are also platform-initiated restarts that happen out of user control)"

In other words. Use Azure Resource Management API to update a value in the site's configuration. (Use a dummy value.)
This triggers a "proper" restart where all application settings are evaluated. 

I've tested this and it works perfectly.

**Comment by v-anvari** [comment]
Thankyou, @23min for the updated information.

@balag0 , Can we add this as a feature request or close this issue for now as this is by-design

**Comment by 23min** [comment]
@balag0 @v-anvari
It would be helpful at least to have clear documentation about various start/stop behaviors (portal, API, platform events), cold start, warm start, and how scaling affects state and configuration. There is information spread around but it is far from intuitive that a Stop/Start does not bring the app up in a clean state.

**Comment by balag0** [comment]
@23min Good to know you were able to workaround this issue.

Regarding start/stop - We identified a race condition bug recently which can sometimes skip the step where currently running workers were not getting recycled correctly. I suspect that could have been the reason for the changes not getting picked up correctly.

In general, any write operations to the site will force a recycle of existing workers and agree that is the best approach here.

**Comment by v-anvari** [comment]
Closing this issue as the required information is provided, please feel free to reach back to us in case of any further queries related to this issue
